### PR TITLE
RSDK-5481: Reduce surface area of log creation APIs.

### PR DESCRIPTION
**Pull request by dgottlieb**
The motivation here is that if i have to implement a new logger, I'd like to match the log output. Zap has a very rich set of customization that would be expensive to keep parity with. Thankfully we only use a small subset of it. This patch/refactor removes unused (or lightly used) code and consolidates the configuration down to (nearly) a single spot.

The [first commit](https://github.com/viamrobotics/rdk/pull/3136/commits/af3ccd8fcbb848214fb0690ad91fe475c0ea5232) actually makes changes. The second only moved code within the logging files. It should be sufficient to just review the first commit and sanity check the latter.

Changes (that I hope to remember to add to the commit message):
* `ReplaceGoabl` -> `ReplaceGlobal`
* The `New*Logger` APIs have been reduced to `NewLogger` and `NewDebugLogger`
  * Strictly speaking, there's some behavior changes due to this that I deemed inconsequential. See below for more details.
* Remove GPC (google cloud provider) loggers. Those were from golog and only used from `viam-robotics/app`.
* Removed Panic/DPanic APIs from Logger.
  * There was one usage of `logger.Panic` and it was confusing to say the least. The behavior was largely retained by separating logger calls + panic call.

Re: Non-GPC `golog.New*Logger` methods, there were 4 public APIs mapping to 3 distinct configurations.
* NewLogger -> NewProductionLogger
* NewProductionLogger
* NewDevelopmentLogger
* NewDebugLogger

The Development and Debug logger were exactly the same except for log level (info vs debug)

The "ProductionLogger" logged at info and had a few differences:
* Used [lower-case uncolored](https://github.com/viamrobotics/rdk/blob/9cde18d23d8a92cc4b6663a17e5305718371ef6c/logging/logger.go#L149) text for error levels in logs ([debug setting](https://github.com/viamrobotics/rdk/blob/9cde18d23d8a92cc4b6663a17e5305718371ef6c/logging/logger.go#L213))
* Encoded [time differently](https://github.com/viamrobotics/rdk/blob/9cde18d23d8a92cc4b6663a17e5305718371ef6c/logging/logger.go#L150) vs [debug](https://github.com/viamrobotics/rdk/blob/9cde18d23d8a92cc4b6663a17e5305718371ef6c/logging/logger.go#L214)
* And [durations](https://github.com/viamrobotics/rdk/blob/9cde18d23d8a92cc4b6663a17e5305718371ef6c/logging/logger.go#L151) [differently](https://github.com/viamrobotics/rdk/blob/9cde18d23d8a92cc4b6663a17e5305718371ef6c/logging/logger.go#L215)

The main viam server only used [development/debug](https://github.com/viamrobotics/rdk/blob/9cde18d23d8a92cc4b6663a17e5305718371ef6c/web/server/entrypoint.go#L71-L75) loggers. There were only a handful of "production logger" instances. Relegated to some CLI programs and a few log lines in other code calling `NewLogger`. There wasn't a lot of meaningful log content in those usages, so I've deemed the usages incidental to using a "production logger" and not intentional.

### CODE REVIEW

File: web/server/entrypoint.go [hunk 1]
```diff
@@ -309,9 +307,8 @@ func (s *robotServer) serveWeb(ctx context.Context, cfg *config.Config) (err err
 				if errors.Is(err, context.Canceled) {
 					return
 				}
-				s.logger.Panicw("error creating restart checker", "error", err)
```
Comment (dgottlieb) [inline web/server/entrypoint.go:312]: This was super weird:
* We're panicing inside of a "PanicCapturingGo"
* "After" the panic we cancel a context.

File: logging/logger.go [hunk 2]
```diff
@@ -100,106 +99,28 @@ var _ Logger = &zLogger{}
...
 )
 
-func newDefaultLogger() Logger {
-	return &zLogger{zap.Must(NewDebugLoggerConfig().Build()).Sugar()}
-}
-
 // ReplaceGloabl replaces the global loggers and returns a function to reset
 // the loggers to the previous state.
-func ReplaceGloabl(logger Logger) func() {
+func ReplaceGlobal(logger Logger) {
```
Comment (benjirewis) [inline logging/logger.go]: 🎉

File: web/server/entrypoint.go (hunk 1 above)
Comment (benjirewis) [reply web/server/entrypoint.go:312]: Yeah... good to get rid of these `Panic` logs; it's an odd function that does not seem particularly useful and is prone to weird "orphaned" logic like the context cancelation and return. I'm reminded of `log.Fatal` which notoriously calls `os.Exit(1)` and can surprisingly crash a whole program through the standard logging library.

File: logging/logging.go [hunk 3]
```diff
@@ -2,76 +2,84 @@
 package logging
```
Comment (benjirewis) [inline logging/logging.go:2]: [q] Can you describe what's in `logging/logger.go` and what's in `logging/logging.go` now?

**Review by benjirewis — APPROVED** [review]
Awesome cleanup/simplification here; thanks 🧑‍🔧 . Just some nits/qs.

**Comment by benjirewis** [comment]
And, seems fine to get rid of production logger from what I can tell. The only other difference I can see that I don't think you mentioned is that [`Development` is explicitly set](https://github.com/viamrobotics/rdk/blob/9cde18d23d8a92cc4b6663a17e5305718371ef6c/logging/logger.go#L138) to `false` (though I assume that's the default value anyway).

File: logging/logging.go (hunk 3 above)
Comment (dgottlieb) [reply logging/logging.go:2]: Sure, sorry for not doing that. Perfectly reasonable for that explanation to have accompanied this PR. Things are still in flux as I settle on code organization. But where I stand now:
* logger.go
  * contains the `Logger` interface
  * contains `ZapCompatibleLogger` interface
  * contains `zLogger` wrapper for ZapCompatible->Logger upconvert
* logging.go
  * Contains the public "static" methods for creating loggers/configurations
 * level.go you didn't ask about
   * But it contains the log Level stuff. Mostly just serialization
 * impl.go does not exist yet in this PR
   * but that's where I'm putting the new Logger implementation that does the things we want (like the upcoming `SetLevel`)
   * I know its customary to make an `impl` directory. If `impl.go` gets large/tangly enough to warrant extra files, I'll go towards a separate directory.

**Comment by dgottlieb** [comment]
> And, seems fine to get rid of production logger from what I can tell. The only other difference I can see that I don't think you mentioned is that [Development is explicitly set](https://github.com/viamrobotics/rdk/blob/9cde18d23d8a92cc4b6663a17e5305718371ef6c/logging/logger.go#L138) to false (though I assume that's the default value anyway).

Yeah I was on the fence of mentioning it given false is the default! But I suppose it was "ok" for a future change to change `Development: true` on the development/debug loggers!

File: logging/logging.go (hunk 3 above)
Comment (benjirewis) [reply logging/logging.go:2]: That all sounds great; thanks!

> I know its customary to make an impl directory.

I haven't heard of this standard in golang (I've always referred to [`golang-standards/project-layout`](https://github.com/golang-standards/project-layout) for "non-official but good" recommendations on project layouts, and they do not use `impl` directories from what I can tell). In any case, `impl` directory and/or `impl.go` seem reasonable.

**Pull request merged**